lib/Command/Room/TRoomCommand.php

```diff
@@ -335,4 +304,60 @@ protected function removeRoomModerators(Room $room, array $userIds): void {
 			$room->setParticipantType($participant, Participant::USER);
 		}
 	}
+
+	protected function completeTokenValues(CompletionContext $context): array {
+		/** @var Manager $roomManager */
+		$roomManager = \OC::$server->query(Manager::class);
+		return array_map(function (Room $room) {
+			return $room->getToken();
+		}, $roomManager->searchRoomsByToken($context->getCurrentWord()));
```

Comment:
```suggestion
		}, $this->manager->searchRoomsByToken($context->getCurrentWord()));
```
And just move
```php
	/** @var Manager */
	public $manager;
```

from all classes to the trait.